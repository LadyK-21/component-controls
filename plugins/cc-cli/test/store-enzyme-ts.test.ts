import path from 'path';
import { runTests } from './run-store-tests';

runTests({
  format: 'ts',
  renderer: 'enzyme',
  config: path.resolve(__dirname, '.config'),
});
