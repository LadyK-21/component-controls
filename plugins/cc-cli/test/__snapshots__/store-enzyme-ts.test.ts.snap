// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme  ts 1`] = `
"import * as path from 'path';
import MatchMediaMock from 'jest-matchmedia-mock';
import {
  loadConfigurations,
  extractDocuments,
} from '@component-controls/config';
import { renderExample } from '@component-controls/test-renderers';
import { render as reactRender } from '@component-controls/render/react';
import { mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

configure({ adapter: new Adapter() });

describe('component-controls generated', () => {
  let matchMedia: MatchMediaMock;
  beforeAll(() => {
    jest.mock('rc-util/lib/Portal');
    matchMedia = new MatchMediaMock();
  });
  afterEach(() => {
    matchMedia.clear();
  });
  const configPath = path.resolve(__dirname, '.config');
  const config = loadConfigurations(configPath);
  if (!config.renderFn) {
    config.renderFn = reactRender;
  }
  const documents = extractDocuments({ config, configPath });
  if (documents) {
    documents.forEach((file: string) => {
      const exports = require(file);
      const doc = exports.default;
      const examples = Object.keys(exports)
        .filter(key => key !== 'default')
        .map(key => exports[key]);

      if (examples.length) {
        describe(doc.title, () => {
          examples.forEach(example => {
            it(example.name, async () => {
              const rendered = renderExample({
                example,
                doc,
                config,
              });

              const component = mount(rendered);
              expect(toJson(component, { mode: 'deep' })).toMatchSnapshot();
            });
          });
        });
      }
    });
  }
});
"
`;
